### DISCUSSION TITLE
Handling AuthStore of 2 different apps hosted on the same domain

### DISCUSSION THREAD

**Discussion by matanshamshila** [pocketbase/pocketbase#4823] [Q&A] [answered]
Hi,
I have 2 different apps built with PocketBase that I host on the same domain (the same users using both apps). As you know, after a successful user sign-in, PB saves an authentication cookie with the key "pb_auth". The issue is when a user signs in to app1, if they then access the URL of app2, the frontend of app2 treats them as an authenticated user even though the AuthStore cookie belongs to app1.

Another issue is that I can't simultaneously hold the AuthStore of two different apps (because the cookie key is always "pb_auth", and I can't use different name). If a user wants to sign in to app2, they first have to sign out from app1.

Would appreciate any suggestions on optional solutions

**Comment by ganigeorgiev** [comment]
> As you know, after a successful user sign-in, PB saves an authentication cookie with the key "pb_auth". 

No, neither PocketBase, nor the JS SDK saves/sets any cookies.

Since you've mentioned cookies I assume that you are following the [JS SSR guide](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#ssr-integration). **The cookie is just a convenient mechanism to share the auth state between the client/browser and your Node.js app.**

If you are using the `exportToCookie` helper you can specify an explicit `path` to target your application, assuming that they are uploaded under a subpath (default to `/`; _see also [Cookies - Path attribute](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#path_attribute)_).

```js
pb.authStore.exportToCookie({ path: "/yourapp" })
```

If you want to export the cookie with a different key, pass a second argument:

```js
pb.authStore.exportToCookie({ path: "/yourapp" }, "your_unique_key")
```

To load a cookie with a custom key, specify it as a second argument of the `loadFromCookie` method:

```js
pb.authStore.loadFromCookie(cookie, "your_unique_key")
```

> Would appreciate any suggestions on optional solutions

I'm repeating my self, but if you are new to web development I don't recommend using JS SSR with PocketBase because most likely you'll mess up the security. PocketBase is designed to be interact directly and the easiest and most performant way to use it is by communicating with the PocketBase server from the client-side. You can read some ramblings on this in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032.

Alternatively, you may not even need PocketBase because if you are using a JS SSR framework you already have a backend (aka. the node server).

If the above doesn't help, please elaborate more clearly on your setup, how your 2 apps work and how do you initialize the SDK to understand the broader picture.

**Reply by ganigeorgiev** [reply, marked as answer]
If you are not using JS SSR and the SDK is initialized on the client-side, then by default `pb.authStore` uses LocalStorage. To set a custom storage key you can initialize it manually:

```js
import PocketBase, { LocalAuthStore } from "pocketbase"

const pb = new PocketBase("http://127.0.0.1", new LocalAuthStore("your_unique_app_key"))
```

**Reply by matanshamshila** [reply]
Thanks for the answer
I might have not been clear enough, but I'm not using JS SSR or other advanced configuration, I simply initialize the SDK on the client-side. Thanks, it works, this is exactly what I was looking for.

Just a quick question: Is it possible to use LocalAuthStore if I install the SDK from the UMD script? Or do I need to install the SDK using ES modules to use LocalAuthStore?

**Reply by ganigeorgiev** [reply]
If you are using the SDK on the client-side, then there are no cookies.

> Is it possible to use LocalAuthStore if I install the SDK from the UMD script?

No, you need to use the ES module (all modern browsers support it as long as the script tag has `type="module"`).
